src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Snapshot/Dependency.cs

```diff
@@ -122,8 +123,70 @@ public Dependency(IDependencyModel dependencyModel)
 
         public string? FilePath { get; }
 
-        public ImageMoniker Icon         => DiagnosticLevel == DiagnosticLevel.None ? Implicit ? IconSet.ImplicitIcon         : IconSet.Icon         : IconSet.UnresolvedIcon;
-        public ImageMoniker ExpandedIcon => DiagnosticLevel == DiagnosticLevel.None ? Implicit ? IconSet.ImplicitExpandedIcon : IconSet.ExpandedIcon : IconSet.UnresolvedExpandedIcon;
+        public ImageMoniker Icon
+        {
+            get
+            {
+                if (DiagnosticLevel == DiagnosticLevel.None)
+                {
+                    if (Implicit)
+                    {
+                        return IconSet.ImplicitIcon;
+                    }
+
+                    return IconSet.Icon;
+                }
+
+                switch (DiagnosticLevel)
+                {
+                    case DiagnosticLevel.UpgradeAvailable:
+                        return KnownMonikers.OfficeWord2013;
+                    case DiagnosticLevel.Warning:
+                        return IconSet.UnresolvedIcon;
+                    case DiagnosticLevel.Deprecation:
+                        return KnownMonikers.OfficeSharePoint2013;
+                    case DiagnosticLevel.Error:
+                        return IconSet.UnresolvedIcon;
+                    case DiagnosticLevel.Vulnerability:
+                        return KnownMonikers.OfficeExcel2013;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
```

Comment:
To be clear (I know this is a draft) but we should consider adding unresolved/deprecation/vulnerability icon members to `IconSet`. The `Dependency` class applies to all kinds of dependencies, not just packages. If another dependency type used one of these new diagnostic levels, we'd end up showing a NuGet package icon for it if we hard coded them here.

This comment is based on the assumption that we end up with package reference icons with new overlays for the three new states.

For non-package dependencies we would likely re-use the warning/error icons for those diagnostic levels, as we don't expect to display them and therefore wouldn't ask the design team to produce such icons for us.

Brief history detour: These icons used to separate the base icon from the overlay. However in 17.0 the design team wanted finer control over how the overlay appeared, and so baked a flattened form of each combination into its own icon.

Reply:
For each new overlay, you'd need a version of it displayed on top of the NuGet icon. That is, baked into a single image for each overlay.